lib/plugin.js

```diff
@@ -120,14 +120,27 @@ function setCookie(ctx, name, value, maxAge = <%= options.maxAge %>) {
 }
 
 // https://developers.google.com/optimize/devguides/experiments
-function googleOptimize({ experiment }) {
-  if (process.server || !window.ga || !experiment || !experiment.experimentID) {
-    return
-  }
+function googleOptimize({ experiment, $gtm }) {
+  if (process.server || !experiment || !experiment.experimentID) return
 
   const exp = experiment.experimentID + '.' + experiment.$variantIndexes.join('-')
 
-  window.ga('set', 'exp', exp)
+  // Choose method to track experiment and variant
+  <% if (options.eventHandler === 'gtm') { %>
```

Comment:
This seems like the wrong direction to me.  What we are essentially doing is corrupting this module based on some select libraries.  What we should be striving for instead is exposing `exp` so anyone can do whatever they like.  

For example, at work we use Segment now, how would this integrate with that? etc..  I understand ga and gtm are common, but if I had my way I wouldn't have auto-included GA to begin with (call me a purist :P)

I do have more plans related to this, but I've been busy with MasteringNuxt for the past months :(

Reply:
@hecktarzuli I absolutely agree with you. A completely global solution would be much better and we'd be giving the flexibility users require.

Reply:
@hecktarzuli @farzadso  See my newest update, what do you think of dynamic handling of  tracking. This would allow users to define their own function to send tracking events. I tested this with $gtm module and it worked well.